### ISSUE TITLE
Compatibility with electron-compile (with a hack solution)

### ISSUE THREAD

**Issue by carlosgalveias** [lovell/sharp#2428]
OS: Windows 10
Node: 14.15.0
electron-forge: '5.2.4'

When trying to include sharp with latest electron-forge, i noticed that altough sharp installed ok, and worked after npm install, when building in electron it was giving a error saying the module did not exists.

After debugging what was going on , electron-recompile was complaining that: 

> fatal error C1189: #error:  Exception support not detected.       Define either NAPI_CPP_EXCEPTIONS or NAPI_DISABLE_CPP_EXCEPTIONS. (compiling source file ..\src\common.cc) [C:\Users\CARLOS~1\AppData\Local\Temp\electron-packager\win32-x64\bluedarwin-studio-win32-x64\resources\app\node_modules\sharp\build\sharp.vcxproj]

Well i defined that flag in bindings.gyp and it stopped complaining, however was still getting the 'Cannot load module' error.
By debugging what was going on i noticed that what electron-rebuild, does is; 

`
 node-gyp rebuild --target=8.2.0 --arch=x64 --dist-url=https://www.electronjs.org/headers --build-from-source
`
If i do that in sharp folder, then node->require('sharp') it also doesn't work.

I noticed then that in sharp package.json it has extra commands : '(node-gyp rebuild && node install/dll-copy)'
so i guessed the last part was breaking it, and it was correct, once i run manually the 'node install/dll-copy' then it worked.

Now to solve that i made a hack, i added;

`'actions': [
      {
      'action_name': 'link_stuff',
      'inputs':[], # complains if no input is provided
      'outputs': ['output.txt'], # complains if not output is provided
      'action': ['node ../install/dll-copy'] # i just want you to run this stuff
      }
    ],`

Into the binding.gyp file, in the 'sharp' target segment.
Basically the 'hack' is for it to run that command upon rebuild as electron-rebuild is not going to run that for me.

And that worked!

Now i have absolutely no idea of what i am doing :)
So i just wanted to leave here my experience so that you guys may do it the right way ;)

**Comment by lovell** [comment]
Native modules that use [N-API](https://nodejs.org/api/n-api.html), such as `sharp`, do not require the use of `electron-rebuild` as the binaries are already suitable for all supported versions of Electron. You'll probably need to report this downstream.

**Comment by carlosgalveias** [comment]
Hi Lovell, thanks for the clarification. I will make a issue downstream.
